optimum/exporters/onnx/model_configs.py

```diff
@@ -192,6 +192,11 @@ class GPTNeoXOnnxConfig(TextDecoderOnnxConfig):
     NORMALIZED_CONFIG_CLASS = NormalizedTextConfig
 
 
+class OPTOnnxConfig(TextDecoderOnnxConfig):
+    DEFAULT_ONNX_OPSET = 13
```

Comment:
Is there any specific reason for this opset?

Reply:
Oh, hi, fxmarty, happy to see you here! So, speaking in general, all transformer models with `LayerNormalization` operator must have opset version equal or higher than 17, its according to https://github.com/onnx/onnx/blob/main/docs/Operators.md. Moreover, after ONNXRuntime graph optimization for transformers, this operator will appear even in models that didn't contain it before optimization, so in my opinion all supported transformer models in optimum must have `DEFAULT_ONNX_OPSET=17` or higher. 

However, in fact, when directly inferencing the model, inappropriate opset is not a big problem, at least in my experience, but checking the ONNX model using `onnx.checker.check_model()` will throw this exception: `ValidationError: No Op registered for LayerNormalization with domain_version of 13`. This might create some pitfalls or confuse people who don't have my level of tolerance for ONNX issues :)

I assigned version 13, simply because it was in some other models in the config file, and in general the model works with it. I can change it to whatever you want here, as you can see.

Reply:
Nice to see you here as well! Thank you for the detail! I hit the exact same issue as you in: https://github.com/microsoft/onnxruntime/issues/14768 My understanding is that the `LayerNormalization` from ONNX Runtime optimization comes actually from `com.microsoft` and not the native ONNX operators. So for the issue with `onnx.checker.check_model()`, I have just ignored it for now as even with opset 17 I was getting `No Op registered` for some other custom ORT nodes.

Leaving opset 13 is fine, I was just being curious!

By the way, off topic but I've had a try at the solution you proposed to use a single decoder by hacking a fake pkv + attention mask. It works nicely. One issue I hit was that some models have a `position_ids` different whether pkv are passed or not, so you actually need this as an input to the ONNX if you want to have the exact same logits (gpt2 and co for example). I want to have a second look at it at some point - it could be the current solution with an If node has some limitations, see: https://github.com/NVIDIA/TensorRT/issues/2723